### PR TITLE
fix: fail when path is not a directory for peercontainer upload/download

### PR DESCRIPTION
**Pull request by kmehant**
Signed-off-by: Mehant Kammakomati <mehant.kammakomati2@ibm.com>

### CODE REVIEW

**Comment by kmehant** [comment]
@seshapad 
@ashokponkumar please review!

File: environment/peercontainer.go [hunk 1]
```diff
@@ -169,6 +169,13 @@ func (e *PeerContainer) Destroy() error {
 
 // Download downloads the path to outside the environment
 func (e *PeerContainer) Download(path string) (string, error) {
+	fileInfo, err := e.Stat(path)
+	if err != nil {
+		return path, fmt.Errorf("failed to stat the given path : %s. Error: %v", path, err)
+	}
+	if !fileInfo.IsDir() {
+		return path, fmt.Errorf("download only supports directory paths. The path provided is %s. Error: %v", path, err)
```
Comment (ashokponkumar) [inline environment/peercontainer.go:177]: Is there a reason to support only directories? Or is our current implementation support only directories?

Comment (seshapad) [reply environment/peercontainer.go:177]: @ashokponkumar From what I understand, I think it is for directories as of now. If you see the containerized transformer code, we actually create a tmp folder to upload the input json.

Having the upload and download path to include files would be a good flexibility to have.

File: environment/peercontainer.go [hunk 2]
```diff
@@ -186,6 +193,13 @@ func (e *PeerContainer) Download(path string) (string, error) {
 // Upload uploads the path from outside the environment into it
 func (e *PeerContainer) Upload(outpath string) (string, error) {
 	envpath := "/var/tmp/" + uniuri.NewLen(5) + "/" + filepath.Base(outpath)
+	fileInfo, err := os.Stat(outpath)
+	if err != nil {
+		return envpath, fmt.Errorf("failed to stat the given path : %s. Error: %v", outpath, err)
+	}
+	if !fileInfo.IsDir() {
+		return envpath, fmt.Errorf("upload only supports directory paths. The path provided is %s. Error: %v", outpath, err)
```
[the remark this answers is not in the document]
Comment (kmehant) [reply environment/peercontainer.go:201]: https://github.com/konveyor/move2kube/issues/1039

**Review by seshapad — APPROVED** [review]

**Pull request merged**